src/common/lib/client/realtimechannel.ts

```diff
@@ -516,7 +512,12 @@ class RealtimeChannel extends EventEmitter {
         if (!message.presence) break;
       // eslint-disable-next-line no-fallthrough
       case actions.PRESENCE: {
-        const presence = message.presence as Array<PresenceMessage>;
+        const presence = message.presence;
```

Comment:
I know it's unrelated to this PR btw but we could definitely improve modularisation by moving this handling out of the realtime channel module, ie:
```ts
switch(message.action) {

  // ...other actions

  case actions.SYNC:
  // eslint-disable-next-line no-fallthrough
  case actions.PRESENCE:
    if (this._presence) {
      this._presence.onMessage(message);
    }
}
```

Reply:
True. I've created https://github.com/ably/ably-js/issues/1516 for it.